packages/ui/react-ui-geo/src/components/Map/Map.tsx

```diff
@@ -2,109 +2,223 @@
-import { useResizeDetector } from 'react-resize-detector';
 
 import { debounce } from '@dxos/async';
 import { ThemeProvider, type ThemedClassName, Tooltip } from '@dxos/react-ui';
 import { defaultTx, mx } from '@dxos/react-ui-theme';
 
+import { type GeoMarker } from '../../types';
 import { ActionControls, type ControlProps, ZoomControls, controlPositions } from '../Toolbar';
-import { type MapCanvasProps } from '../types';
 
 // TODO(burdon): Explore plugins: https://www.npmjs.com/search?q=keywords%3Areact-leaflet-v4
 // TODO(burdon): react-leaflet v5 is not compatible with react 18.
+// TODO(burdon): Guess initial location.
 
 const defaults = {
-  // TODO(burdon): Guess location.
-  center: { lat: 51, lng: 0 } as L.LatLngExpression,
+  center: { lat: 51, lng: 0 } as L.LatLngLiteral,
   zoom: 4,
 };
 
 //
-// Root
+// Controller
 //
 
-type MapRootProps = ThemedClassName<MapContainerProps>;
-
-// https://react-leaflet.js.org/docs/api-map
-const MapRoot = ({ classNames, center = defaults.center, zoom = defaults.zoom, ...props }: MapRootProps) => {
-  return (
-    <MapContainer
-      className={mx('relative grid grow bg-baseSurface', classNames)}
-      attributionControl={false}
-      // TODO(burdon): Only if attention.
-      scrollWheelZoom={true}
-      zoomControl={false}
-      center={center}
-      zoom={zoom}
-      {...props}
-    />
-  );
+type MapController = {
+  setCenter: (center: LatLngLiteral, zoom?: number) => void;
+  setZoom: (cb: (zoom: number) => number) => void;
 };
 
 //
-// Control
+// Context
 //
 
-// TODO(burdon): Normalize with Globe.
-type MapController = {
-  setCenter: (center: LatLngExpression, zoom?: number) => void;
-  setZoom: (cb: (zoom: number) => number) => void;
+type MapContextValue = {
+  attention?: boolean;
 };
 
-const MapCanvas = forwardRef<MapController, MapCanvasProps>(({ markers, center, zoom, onChange }, forwardedRef) => {
-  const { ref, width, height } = useResizeDetector({ refreshRate: 200 });
-  const map = useMap();
+const [MapContextProvier, useMapContext] = createContext<MapContextValue>('Map');
+
+//
+// Root
+//
+
+type MapRootProps = ThemedClassName<
+  MapContainerProps & {
+    onChange?: (ev: { center: LatLngLiteral; zoom: number }) => void;
+  }
+>;
 
-  useImperativeHandle(
+/**
+ * https://react-leaflet.js.org/docs/api-map
+ */
+const MapRoot = forwardRef<MapController, MapRootProps>(
+  (
+    {
+      classNames,
+      scrollWheelZoom = true,
+      doubleClickZoom = true,
+      touchZoom = true,
+      center = defaults.center,
+      zoom = defaults.zoom,
+      onChange,
+      ...props
+    },
     forwardedRef,
-    () => ({
-      setCenter: (center: LatLngExpression, zoom?: number) => {
-        map.setView(center, zoom);
-      },
-      setZoom: (cb) => {
-        map.setZoom(cb(map.getZoom()));
-      },
-    }),
-    [map],
-  );
+  ) => {
+    const [attention, setAttention] = useState(false);
```

Comment:
@richburdon this is not what I meant when I left [this comment](https://github.com/dxos/dxos/pull/9605#discussion_r2251852708), this just renames ‘focus’ to ‘attention’ which confuses the matter…

I’d recommend at least leaving a to-do to follow-up on this since it’s definitely tech debt.